src/Flexdis86/Assembler.hs

```diff
@@ -109,6 +112,26 @@ findEncoding args def = do
               , iiRequiredRM  = L.view requiredRM  def
               }
 
+-- | Infer whether the given instruction and its arguments imply the use of an
+-- operand-size override.
+--
+-- This definition is woefully incomplete at the moment, and cases have been
+-- added on an as-needed basis thus far. This conservatively returns 'False' for
+-- cases that have not been implemented yet.
+mkOSO :: Def -> [(Value, OperandType)] -> Bool
+mkOSO def argTypes
+    -- In 64-bit mode, the default operand size for `push` is 64 bits. If a
+    -- 16-bit operand is used, it must be due to an operand-size override.
+  | def^.defMnemonic == "push"
+  = case argTypes of
+     [(WordImm{},       _)] -> True
+     [(WordReg{},       _)] -> True
+     [(WordSignedImm{}, _)] -> True
```

Comment:
I had assumed that `flexdis86` was assuming 64-bit mode due to this code:

https://github.com/GaloisInc/flexdis86/blob/9bd6fc6a9480d2ea0cd22d04fd5a5e4ae532504a/src/Flexdis86/Prefixes.hs#L211-L213

If I'm reading [this table](https://wiki.osdev.org/X86-64_Instruction_Encoding#Operand-size_and_address-size_override_prefix) correctly, this will only really work on 64-bit mode.

Reply:
Thanks - this looks good to me